docs/troubleshooting.md

```diff
@@ -0,0 +1,141 @@
+# Troubleshooting
+
+This document focuses on troubleshooting unexpected behavior when installing and managing operators with OLM. This document will contain 3 sections in which we highlight how to troubleshoot specific OLM components:
+
+* [Troubleshooting the `CatalogSource`](#catalogsource-troubleshooting).
+* [Troubleshooting the `Subscription`](#subscription-troubleshooting).
+* [Troubleshooting the `CatalogServiceVersion (CSV)`](#clusterserviceversion-troubleshooting).
+
+>Note: Some of the commands listed below assume that you have [yq](https://github.com/mikefarah/yq) installed on your system. While `yq` is not required, it is a useful tool when parsing yaml. You can install `yq` by following the [official install steps](https://github.com/mikefarah/yq#install).
+
+## CatalogSource Troubleshooting
+
+### How to debug a failing CatalogSource
+
+The Catalog operator will constantly update the Status of `CatalogSources` to reflect its current state. You can check the status of your `CatalogSource` with the following command:
+
+`$ kubectl -n my-namespace get catsrc my-catalog -o yaml | yq r - status`
+
+>Note: It is possible that the `Status` is missing, which suggests that the Catalog operator is encountering an issue when processing the `CatalogSource` in a very early stage.
+
+If the status block does not provide enough information, check the [Catalog operator's logs](#how-to-view-the-catalog-operator-logs).
+
+If you are still unable to identify the cause of the failure, check if a pod was created for the `CatalogSource`. If a pod exists, review the pod's yaml and logs:
+
+```sh
+$ kubectl -n my-namespace get pods
+NAME                                READY   STATUS    RESTARTS   AGE
+my-catalog-ltdlp         1/1     Running   0          8m31s
+
+$ kubectl -n my-namespace get pod my-catalog-ltdlp -o yaml
+...
+
+$ kubectl -n my-namespace logs my-catalog-ltdlp
+...
+```
+
+### I'm not sure if a CatalogSource contains an Operator I want to install
+
+First verify that the `CatalogSource` contains the operator that you want to install:
+
+```sh
+$ kubectl -n my-namespace get packagemanifests
+NAME                               CATALOG             AGE
+...
+portworx                           My Catalog Source   14m
+postgres-operator                  My Catalog Source   14m
+postgresql                         My Catalog Source   14m
+postgresql-operator-dev4devs-com   My Catalog Source   14m
+prometheus                         My Catalog Source   14m
+...
+```
+
+If the opreator is not listed, rebuild the CatalogSource with the [operator-sdk bundle command](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#operator-bundle-commands).
```

Comment:
A few thoughts:

The operator-sdk doesn't actually have any of these commands yet. Even then,the registry images built with the bundle aren't actually available yet. Creating them involves compiling `opm` and building your own image. Also, building the bundle and building the index image are different -- the catalogsource doesn't get data from bundle images themselves, it gets them from the index which was built from the bundles. That second step won't ever be based on the doc you are linking here since `opm index` is never going to get added to the `operator-sdk`.

Do we want to even mention recreating a catalog here? It seems like if the audience is "my operator isn't listed" then they probably already know how to build a catalog themselves. If the audience is "I don't know that creating a catalog with a specific operator in it is how i get that operator" then they probably don't have the operator to add to it anyway. Maybe in the future when additive catalogs are easy to use and there are lots of common published docs about this we can make it easy to roll your own with publicly available bundles. But for now I think we should maybe just leave it as a TODO?

Reply:
This is fair - I have removed the related bits.